app/internal/windows/windows.go

```diff
@@ -814,3 +922,71 @@ func (p *WindowPlacement) Set(Left, Top, Right, Bottom int) {
 	p.rcNormalPosition.Right = int32(Right)
 	p.rcNormalPosition.Bottom = int32(Bottom)
 }
+
+func SearchHWND(path string) (syscall.Handle, bool) {
+	base := filepath.Base(path)
+
+	var resHWND syscall.Handle
+	fn := syscall.NewCallback(func(hwnd uintptr, lparam uintptr) uintptr {
+		var processID uint32
+		_GetWindowThreadProcessID.Call(hwnd, uintptr(unsafe.Pointer(&processID)))
+
+		handle, _ := syscall.OpenProcess(syscall.PROCESS_QUERY_INFORMATION|syscall.PROCESS_VM_READ, false, processID)
+		if handle != 0 {
+			defer syscall.CloseHandle(handle)
+
+			var buffer [4096]uint16
+			size := uint32(len(buffer))
+			_QueryFullProcessImageName.Call(uintptr(handle), 0, uintptr(unsafe.Pointer(&buffer)), uintptr(unsafe.Pointer(&size)))
+
+			if base == filepath.Base(syscall.UTF16ToString(buffer[:size])) {
+				resHWND = syscall.Handle(hwnd)
+				return 0
+			}
+		}
+
+		return 1
+	})
+
+	_EnumWindows.Call(fn, 0)
+	return resHWND, resHWND != 0
+}
+
+func RegisterScheme(scheme string) error {
```

Comment:
Can you explain how `gogio` can improve that? I mean, how can we "bundle" two apps or set such Registry.

I don't like the `RegisterScheme` too, and the "IPC" thing. But, considering that Gio tries to map features across each OS as similar as possible, forcing Windows to have a installer doesn't match against macOS. Also, `gogio` only emits a single binary (`.exe`), and don't generate other installers (such as `.appx` or `.msix`).

On macOS a single `.app` is responsible to register itself, due to `plist`. On Windows, such registration is done by calling `RegisterScheme`, and then we need build something to send messages.

I agree that using a Gio executable to register and also send messages is not ideal. But, that seems to be the closest behaviour (vs macOS).

Reply:
Ok, I suppose this code can stay, because it is not triggered without `gogio` embedding one or more schemes.

Can we clean up registries no longer needed, to mimic the macOS plist behaviour? In other words, have the registry store the `appid` or whatever you need to identify windows belonging to the same process, and then delete schemes no longer included in `schemesURI`.